tests/test_vqe.py

```diff
@@ -886,6 +887,38 @@ def circuit(params):
         mt2 = circuit.metric_tensor([p])
         assert np.allclose(mt, mt2)
 
+    def test_multiple_devices(self, mocker):
+        """Test that passing multiple devices to ExpvalCost works correctly"""
+
+        dev = [qml.device("default.qubit", wires=2), qml.device("default.mixed", wires=2)]
+        spy = mocker.spy(DefaultQubit, "apply")
+        spy2 = mocker.spy(DefaultMixed, "apply")
+
+        h = qml.Hamiltonian([1, 1], [qml.PauliZ(0), qml.PauliZ(1)])
+
+        qnodes = qml.ExpvalCost(qml.templates.StronglyEntanglingLayers, h, dev)
+        w = qml.init.strong_ent_layers_uniform(3, 2, seed=1967)
+
+        res = qnodes(w)
+        exp = 0.3864767030806935
+
+        spy.assert_called_once()
+        spy2.assert_called_once()
+
+        assert np.allclose(res, exp)
+
+    def test_multiple_devices_opt_true(self):
+        """Test if a ValueError is raised when multiple devices are passed when optimize=True."""
+        if not qml.tape_mode_active():
+            pytest.skip("This test is only intended for tape mode")
+
+        dev = [qml.device("default.qubit", wires=2)] * 2
```

Comment:
(Minor suggestion): A subtlety here is that we are passing a list of two references to the same device, maybe worth checking that this works well with two separate devices?
```suggestion
        dev = [qml.device("default.qubit", wires=2), qml.device("default.qubit", wires=2)]
```
---
```python
In [14]: b = [qml.device("default.qubit", wires=2)] * 2                         

In [15]: b[0] is b[1]                                                           
Out[15]: True
```

Reply:
Thanks, added!